robottelo/api/utils.py

```diff
@@ -130,6 +133,38 @@ def publish_puppet_module(puppet_modules, repo_url, organization_id=None):
     return cv.read()
 
 
+def import_puppet_module(sat, puppet_module_repo=CUSTOM_PUPPET_MODULE_REPOS['generic_1']):
+    """Download, install and import puppet module. This is workaround for 6.10, there is no longer
+    content view
+
+    :param puppet_module_repo: custom puppet module repository
+    :param sat: Satellite object
+    :return: Puppet environment name,
+        environment name is likely to be searched in next steps of the test
+    """
+    if puppet_module_repo not in list(CUSTOM_PUPPET_MODULE_REPOS.values()):
+        raise ValueError(
+            f'Custom puppet module mismatch, actual custom puppet module repo: '
+            f'"{puppet_module_repo}", does not match any available puppet module: '
+            f'"{list(CUSTOM_PUPPET_MODULE_REPOS.values())}"'
+        )
+    env_name = gen_string('alpha')
+    custom_puppet_module_repo = f'{puppet_module_repo}{CUSTOM_PUPPET_MODULE_REPOS_VERSION}'
+    sat.execute(
+        f'curl -O {settings.robottelo.repos_hosting_url}{CUSTOM_PUPPET_MODULE_REPOS_PATH}'
+        f'{custom_puppet_module_repo}',
+    )
+    sat.execute(
+        f'puppet module install {custom_puppet_module_repo} '
+        f'--target-dir /etc/puppetlabs/code/environments/{env_name}/modules/'
+    )
+    smart_proxy = (
+        entities.SmartProxy().search(query={'search': f'name={settings.server.hostname}'})[0].read()
+    )
+    smart_proxy.import_puppetclasses()
```

Comment:
https://apidocs.theforeman.org/foreman/latest/apidoc/v2/environments/import_puppetclasses.html suggests you can run it only for a single environment, but maybe then the environment must already exist and it's the DB ID. Overall it would be cleaner to only import the environment you touched, to prevent accidentally breaking other tests.